src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -154,72 +181,162 @@ export default class SimidMapCreative extends BaseSimidCreative {
+   * and places pins on the map that represent the 4 closest locations.
+   * @param {String} searchParameter A string with the advertiser's name to use in the query.
+   * @param {!google.maps.LatLng} coordinates The LatLng object of user's current location.
+   * @private 
+   */
+  findNearby_(searchParameter, coordinates) {
+    const request = {
+      location: coordinates,
+      name: searchParameter,
+      openNow: true,
+      rankBy: google.maps.places.RankBy.DISTANCE
+    };
+    const service = new google.maps.places.PlacesService(this.map_);
+    service.nearbySearch(request, this.displayResults_.bind(this));
+    this.createTravelChoices_();
+  }
 
-/**
- * Displays the closest business locations to a user's current location.
- * @param {!Object} results An array of Place Results from the search query.
- * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
- *  by the PlacesService on the completion of its searches.
- * @private 
- */
-displayResults_(results, status) {
+  /**
+   * Displays the closest advertisement's locations to a user's current location.
+   * @param {!Object} results An array of Place Results from the search query.
+   * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+   *  by the PlacesService on the completion of its searches.
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.activeLocation_ = results[0].geometry.location;
+      this.displayDirections_(this.activeLocation_, this.currentLocation_);
+    } else {
+      const statusErrorMessage = {
+        message: "ERROR: Places Service Status was: "+status,
+      };
+      this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+      this.playContent_();
     }
-}
+  }
 
-/**
- * Creates and displays a marker on the map representing a given place.
- * @param {!Object} place A Place Result object.
- * @private 
- */
-placeMapMarker_(place) {
-  const placeIcon = {
-    url: this.markerImage_,
-    scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-  };
-  const placeMarker = new google.maps.Marker({
-    map: this.map_,
-    position: place.geometry.location,
-    icon: placeIcon
-  });
-}
+  /**
+   * Creates and displays a marker on the map representing a given place.
+   * @param {!Object} place A Place Result object.
+   * @private 
+   */
+  placeMapMarker_(place) {
+    const placeIcon = {
+      url: this.markerImage_,
+      scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
+    };
+    const placeMarker = new google.maps.Marker({
+      map: this.map_,
+      position: place.geometry.location,
+      icon: placeIcon
+    });
+    ///Recalculate directions if new marker is clicked.
```

Comment:
what does this mean?

Reply:
If a different marker is selected on the map, it becomes the new active marker & the directions will be recalculated accordingly. Updated the comment to hopefully be a little more clear, "Recalculate directions if a different active marker is selected."